draft-ietf-mls-protocol.md

```diff
@@ -4993,11 +5028,11 @@ parent=01x => left=00x, right=10x
 ~~~
 
 Since node relationships are implicit, the algorithms for adding and removing
-nodes at the right edge of the tree are quite simple:
+nodes at the right edge of the tree are quite simple.  If there are `N` nodes in
+the array:
 
-* Add: Append a blank parent node to the array of nodes, then append the new
-  leaf node
-* Remove: Remove the rightmost two nodes from the array of nodes
+* Add: Append `N - 1` blank values to the end of the array.
```

Comment:
```suggestion
* Add: Append `N + 1` blank values to the end of the array.
```

I think it's N+1 because you need the new right tree of the root (N nodes) + 1 node for the root